includes/Core/Authentication/Authentication.php

```diff
@@ -504,6 +531,35 @@ private function handle_oauth() {
 		}
 	}
 
+	/**
+	 * Handles receiving a verification token for a user by the authentication proxy.
+	 *
+	 * @since 1.0.0
+	 */
+	private function handle_verification_token() {
+		$auth_client = $this->get_oauth_client();
+		if ( ! $auth_client->using_proxy() ) {
+			return;
+		}
+
+		$verification_token = filter_input( INPUT_GET, 'googlesitekit_verification_token' );
+		if ( empty( $verification_token ) ) {
+			return;
+		}
+
+		$verification_nonce = filter_input( INPUT_GET, 'googlesitekit_verification_nonce' );
+		if ( empty( $verification_nonce ) || ! wp_verify_nonce( $verification_nonce, 'googlesitekit_verification' ) ) {
+			wp_die( esc_html__( 'Invalid nonce.', 'google-site-kit' ) );
+		}
+
+		$this->verification_tag->set( $verification_token );
+
+		$code = (string) filter_input( INPUT_GET, 'googlesitekit_code' );
```

Comment:
Can we extract these query params to class constants? E.g. `self::PARAM_VERIFICATION_TOKEN` etc